### PR TITLE
Deploy Tool Docs

### PR DESCRIPTION
**Pull request by aaronranard**
Documentation about the Deploy tool and Third Party Tools

### CODE REVIEW

File: docs/deployment/nio/release.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+<img class="left" src="/img/deploy/release/toolbar.png" />
+
+### Creating a new Instance Configuration
+
+This will show the option to create a a new Instance Configuration. Give your Instance Configuration a name and an initial version. We recommend `v1.0.0` which follows the [semver](https://semver.org) scheme.
+
+<img class="left" src="/img/deploy/release/modal.png" height="350" />
+
+Click `OK` to release your first Instance Configuration!
```
Comment (kyliedale) [inline docs/deployment/nio/release.md]: ok should be bold here too to follow our convention

File: docs/deployment/nio/release.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+If you would now like to deploy, you should move on to:
+
+* [Direct Deployment](/deployment/nio/direct.md)
+* [Release Deployment](/deployment/nio/release.md)
```
Comment (kyliedale) [inline docs/deployment/nio/release.md]: Did you mean to link to the page the user is on?

File: docs/deployment/nio/release.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+### Releasing a new Version
+
+If an Instance Configuration is already created and only a new version needs to be released, select the Instance Configuration from the select field and then give the version a new number. Remember, an Instance Configuration Version can be named using any versioning scheme, we recommend [semver](https://semver.org), and must be unique.
+
+<img class="left" src="/img/deploy/release/version.png" height="350" />
+
+Click `OK` to release the new Instance Configuration Version!
```
Comment (kyliedale) [inline docs/deployment/nio/release.md]: bolded here

File: docs/deployment/nio/release.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+---
+
+## Deploy
+
+After successfully releasing an instance configuration
```
Comment (kyliedale) [inline docs/deployment/nio/release.md]: `After successfully releasing an instance configuration move on to either of the following:`?

File: docs/deployment/nio/direct.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
+# Direct Deployment
+
+A Direct Deployment is an instant deployment to instances that **are** externally accessible. 
+
+> **[info] Compatibility**
+>
+> If you can load your instance in the System Designer, you can do a Direct Deployment to it.
```
Comment (kyliedale) [inline docs/deployment/nio/direct.md]: `If you can load your instance in the System Designer, you can deploy directly to it.` ?

File: docs/deployment/nio/direct.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+## Doing a Direct Deployment
+
+> **[danger] Requirements**
+>
+> A [Release](/deployment/nio/release.md) must be done before a deployment can be done.
+>
+
+- To do a direct deployment, navigate to the deployment tool on [app.n.io/deploy](https://app.n.io/deploy).
+- Next click the **Deploy** button in the top right corner.
```
Comment (kyliedale) [inline docs/deployment/nio/direct.md]: lowercase deploy to match ui?

File: docs/deployment/nio/direct.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+<img class="left" src="/img/deploy/direct/button.png" />
+
+---
+
+- Select the System with which you would like to deploy to.
```
Comment (kyliedale) [inline docs/deployment/nio/direct.md]: you don't need to capitalize 'system' here

File: docs/deployment/nio/direct.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+- You will then be prompted with the new Deployment Page:
+
+<img src="/img/deploy/direct/new.png" />
+
+---
+
+- Notice the instances on bottom. They need to be assigned to the configuration that should be deployed.
```
Comment (kyliedale) [inline docs/deployment/nio/direct.md]: `at the bottom`

File: docs/deployment/nio/direct.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+---
+
+- The next step is to select the correct version to deploy for each configuration.
+
+<img src="/img/deploy/direct/version.png" />
+
+---
+
+- After that is done, review the deployment and if everything looks correct click `deploy`!
```
Comment (kyliedale) [inline docs/deployment/nio/direct.md]: bold 🙂

File: docs/deployment/nio/direct.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+- The deployment will begin and will report it's progress when complete.
```
Comment (kyliedale) [inline docs/deployment/nio/direct.md]: `its`

File: docs/deployment/nio/direct.md [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+- After that is done, review the deployment and if everything looks correct click `deploy`!
+- The deployment will begin and will report it's progress when complete. 
+
+<img src="/img/deploy/direct/complete.png" />
+
+---
+
+## Deployment Details
+
+- The deployment can be clicked on to view specific details about each instance that was deployed to. For instance, if the deployment is reporting as failed, you can click in and investigate which instance in particular may have caused the failure.
```
Comment (kyliedale) [inline docs/deployment/nio/direct.md]: `if the deployment reported as failed`

File: docs/deployment/nio/component.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
+# Component Deployment
+
+A Component Deployment is a deferred deployment to instances that **are not** externally accessible, or a component that should self update once a new version is released. This utilizes the [nio config component](https://github.com/niolabs/component_config).
+
+> **[info] Designer**
+>
+> If your instance is not externally accessible, design the configuration and [do a release](/deployment/nio/release.md) in the [nio System Designer](https://app.n.io/design) using a [Cloud Instance](/quickstart/README.md) or an accessible local instance.
```
Comment (kyliedale) [inline docs/deployment/nio/component.md]: cloud instance doesn't need to be capitalized here

File: docs/deployment/nio/component.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+>
+
+## Doing a Component Deployment
+
+> **[danger] Requirements**
+>
+> An initial [Release](/deployment/nio/release.md) must be done before continuing further.
```
Comment (kyliedale) [inline docs/deployment/nio/component.md]: I would keep this consistent with the other one: 
```
must be done before a deployment can be done
```

File: docs/deployment/nio/component.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+>
+
+#### To do a component deployment:
```
Comment (kyliedale) [inline docs/deployment/nio/component.md]: Each bullet in the following list should have a period I think

File: docs/deployment/nio/component.md [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+- Make sure your nio instance has the [nio config component](https://github.com/niolabs/component_config)
+- Get the Instance Configuration ID from the Release modal in the System Designer.
+
+<img src="/img/deploy/component/id.png" height="350" />
+
+- Copy the `id` value for the configuration you would like to use.
```
Comment (kyliedale) [inline docs/deployment/nio/component.md]: `id` should be bold

File: docs/deployment/nio/release.md (hunk 2 above)
Comment (aaronranard) [reply docs/deployment/nio/release.md]: Definitely not. Great catch!

File: docs/deployment/README.md [hunk 16]
```diff
@@ -1,26 +1,11 @@
-# Deployment
+# Instance Deployment
 
-Designing a stable nio system means having reliable mechanisms for deployment and maintenance. The deployment of nio systems is very flexible, but there are a few best practices to follow.
+Designing a stable nio system means having reliable mechanisms for deployment and maintenance. The deployment of nio systems is very flexible. 
 
----
-## `git` for version control
+## nio Deploy Tool
```
Comment (kyliedale) [inline docs/deployment/README.md]: We should capitalize `Deploy Tool` here too with the <span class=...

**Pull request merged**